TACC/protx-dashboard
# task/COOKS-303: fix json response speed

## Overview: ##

This PR fixes an issue where converting a dict to json string using flask approach is very slow as the flask approach is to "prettify" the json which doesn't scale well for our large dicts/json. In this PR, we handle the dict->json conversion and also compress the result in our cache instead of the raw json.

This PR also does the following:
* Improves logging a bit
* Update our "run_queries" script so that cache info is cleared before running 

## Related Jira tickets: ##

* [COOKS-303](https://jira.tacc.utexas.edu/browse/COOKS-303)

## Testing Steps: ##
1. Start up as usual and ensure you run `docker exec -it protx python3 scripts/run_queries.py --clear-cache`.
2. Load page in dev mode and confirm that `protx/api/demographics` and `protx/api/resources` loads faster (  "waiting on server" is just around a second long and then downloading begins. previously it would take 10-30 seconds before downloading began).

## Notes ###

TODO 
- [x] failing
- [x] fix or add a ticket for failing unit tests
